Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ContextMenu.cpp

```diff
@@ -461,10 +512,44 @@ namespace EMStudio
                 }
             }
 
-            if (actionFilter.m_editNodeGroups &&
-                !inReferenceGraph)
+            bool all_nodes_are_ungrouped = std::all_of(
+                selectedNodes.cbegin(),
+                selectedNodes.cend(),
+                [&](EMotionFX::AnimGraphNode* node)
+                {
+                    return node->GetAnimGraph()->FindNodeGroupForNode(node) == nullptr;
+                });
+            if (all_nodes_are_ungrouped)
+            {
+                QAction* createNodeGroupAction = menu.addAction("Create Node Group");
+                connect(createNodeGroupAction, &QAction::triggered, this, &BlendGraphWidget::CreateNodeGroup);
+                if (actionFilter.m_editNodeGroups && !inReferenceGraph)
+                {
+                    AddAssignNodeToGroupSubmenu(&menu, selectedNodes[0]->GetAnimGraph(), selectedNodes);
+                }
+            }
+            else
             {
-                AddNodeGroupSubmenu(&menu, selectedNodes[0]->GetAnimGraph(), selectedNodes);
+                QAction* removeFromGroupAction = menu.addAction("Remove From Node Group");
+                connect(
+                    removeFromGroupAction,
+                    &QAction::triggered,
+                    this,
+                    [=]
+                    {
+                        for (EMotionFX::AnimGraphNode* node : selectedNodes)
+                        {
+                            auto group = node->GetAnimGraph()->FindNodeGroupForNode(node);
```

Comment:
Is using `auto` here okay? Other code in the repository uses it, but not the code nearby here.

Reply:
Yeah I think this is fine, I think it's useful for casts especially (e.g. `auto thing = static_cast<int>(otherThing);`) as then the redundancy is removed but even in cases like this it's not bad. One suggestion is if this is a pointer, we tend to prefer `auto*` as opposed to just `auto` to make that clearer to the reader but it's not a massive deal.